### ISSUE TITLE
No map display with style rule for NULL value 

### ISSUE THREAD

**Issue by Jeroen-GroeneBij** [tomchadwin/qgis2web#533]
Hello,

I am using Qgis 2.14.16 and qgis2web 2.23.1
I have made a map with a rule based style for an attribute. When I add a rule for NULL values the qgis2web plugin does not render the map, both in Leaflet end Openlayer.
The rule is configured as:
"Attributename" IS NULL
This does work nicely in qgis, presenting all the polygons in the selected colour with no value for that specific attribute. The qgis2web plugin doesn't seem to like this rule.

The rule itself however doesn't seem to be the problem: when every field has a value, and the NULL rule exists, the qgis2web plugin does render the map. As soon as there is a field with no value (NULL) for that specific attruibute, qgis2web does not render a map.
If there is a field with no value, but the rule is deleted, qgis2web again does render the map.

So the combination of an IS NULL rule and a field with no value results in a problem with the plugin.

I hope you have time to look into this.

**Comment by tomchadwin** [comment]
I've committed a change which might help with this, but I'm not sure I've got it right. The test I made ended up with the value `None` coming out of QGIS, not `null`. Can you test master and see if it works?

@NathanW2 Does this fix look right to you?

**Comment by Jeroen-GroeneBij** [comment]
I tried to make the changes in exp2js.py using notepad++ (deleting line 183 and adding the new lines 183,184 and 185).
But now the plugin does not load, probably some mistake I made? So I can't test master it.

**Comment by tomchadwin** [comment]
Open the qgis2web tab in the QGIS log messages panel (in QGIS, View > Panels > Log messages panel). See if there are errors there you can paste here.

**Comment by Jeroen-GroeneBij** [comment]
Traceback (most recent call last):
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 306, in startPlugin
			    plugins[packageName] = package.classFactory(iface)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\__init__.py", line 28, in classFactory
			    from .qgis2web import Qgis2Web
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\qgis2web.py", line 29, in <module>
			    from maindialog import MainDialog
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\maindialog.py", line 70, in <module>
			    from olwriter import OpenLayersWriter
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\olwriter.py", line 47, in <module>
			    from olStyleScripts import exportStyles
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\olStyleScripts.py", line 23, in <module>
			    from exp2js import compile_to_file
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\exp2js.py", line 183
			    if val is None
			                 ^
			SyntaxError: invalid syntax

**Comment by tomchadwin** [comment]
Do you have the colon at the end?

**Comment by Jeroen-GroeneBij** [comment]
My mistake.
This is the error message with the colon.

Traceback (most recent call last):
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 306, in startPlugin
			    plugins[packageName] = package.classFactory(iface)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\__init__.py", line 28, in classFactory
			    from .qgis2web import Qgis2Web
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\qgis2web.py", line 29, in <module>
			    from maindialog import MainDialog
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\maindialog.py", line 70, in <module>
			    from olwriter import OpenLayersWriter
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\olwriter.py", line 47, in <module>
			    from olStyleScripts import exportStyles
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\olStyleScripts.py", line 23, in <module>
			    from exp2js import compile_to_file
			  File "C:/PROGRA~1/QGIS2~1.14/apps/qgis-ltr/./python\qgis\utils.py", line 572, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "C:/Users/Jeroen/.qgis2/python/plugins\qgis2web\exp2js.py", line 185
			    elif isinstance(val, basestring):
			                                    ^
			IndentationError: unindent does not match any outer indentation level

**Comment by tomchadwin** [comment]
OK, you need to make sure that the indentation is correct: lines 183 and 185 are preceded by 4 spaces, and line 184 by 8.

**Comment by Jeroen-GroeneBij** [comment]
Got it, used a tab instead of spaces. I am new to coding :)

Now the plugin does render the map, but does not show the specific polygon with the Null value.
I tried changing line 184 "null" to "Null' but then the map did not render at all.

**Comment by tomchadwin** [comment]
OK. Export the map, and then hit F12 in your browser, then click on the Console tab. Paste here any errors you find there.

**Comment by Jeroen-GroeneBij** [comment]
There is an error within the  CSS part of the Console Tab:
Some of this is in Dutch...

Onbekende eigenschap ‘user-select’.  Declaratie genegeerd.  leaflet.css:25:20
Fout tijdens het parsen van waarde voor ‘image-rendering’.  Declaratie genegeerd.  leaflet.css:30:18
Onbekende eigenschap ‘behavior’.  Declaratie genegeerd.  leaflet.css:105:9
Fout tijdens het parsen van waarde voor ‘cursor’.  Declaratie genegeerd.  leaflet.css:206:9
Fout tijdens het parsen van waarde voor ‘cursor’.  Declaratie genegeerd.  leaflet.css:221:9
Onbekende eigenschap ‘zoom’.  Declaratie genegeerd.  leaflet.css:527:5
‘progid’ gevonden waar ‘not’, URL of filterfunctie verwacht.  Fout tijdens het parsen van waarde voor ‘filter’.  Declaratie genegeerd.  leaflet.css:534:9
Onbekende eigenschap ‘user-select’.  Declaratie genegeerd.  leaflet.css:569:12
Onbekende eigenschap ‘-moz-border-radius’.  Declaratie genegeerd.  Control.OSMGeocoder.css:10:19

**Comment by tomchadwin** [comment]
No, we need javascript errors. What browser are you using, on which operating system? There should be a separate console which show JS errors.

**Comment by Jeroen-GroeneBij** [comment]
I am using Firefox on win10
Within the Console tab there is a JS section, but it doesn't show anything. No errors.
I could email you the export map?

**Comment by tomchadwin** [comment]
Zip and upload it here if you can.

**Comment by Jeroen-GroeneBij** [comment]
[qgis2web_2017_08_25-17_07_19_997000.zip](https://github.com/tomchadwin/qgis2web/files/1252271/qgis2web_2017_08_25-17_07_19_997000.zip)

**Comment by tomchadwin** [comment]
That's all working correctly. If `Aantal` is null, it uses a plain style: black border, white fill, no transparency. Is that not what your QGIS project specifies?

**Comment by Jeroen-GroeneBij** [comment]
It is. And in qgis the polygon is visible (the black border).
In the webmap the border is not visible, and the entire polygon seems to be missing. I know where it should be, but mouse-over doesn't show it and it doesn't show up with use of the search bar (address search).

**Comment by tomchadwin** [comment]
What value does that polygon have for 'gemeentena'?

**Comment by Jeroen-GroeneBij** [comment]
Onderbanken

**Comment by tomchadwin** [comment]
Interesting. That record in the webmap export doesn't have Aantal at all:

```javascript
,{
"type":"Feature","properties":{
"gemeentena":"Deurne","Aantal":2.2}
,"geometry":{
"type":"Polygon","coordinates":[[[5.874965974060582,51.446557719307457],[5.915798861471014,51.402781107965467],[5.931287597326956,51.384752673544135],[5.878699459706886,51.355532691630813],[5.874543224360317,51.353312722060764],[5.870359646243736,51.352534962614961],[5.866018957997467,51.354620451403065],[5.856580686349925,51.353763914718506],[5.853609183172708,51.369749826077744],[5.853235259043974,51.372728641498831],[5.854103184628019,51.376429627823953],[5.842689992329051,51.390201941994349],[5.839598752641802,51.393153885879151],[5.838798270016226,51.394589691682626],[5.838271052908731,51.395257292653071],[5.837439357510115,51.395927892633296],[5.836437127133215,51.396453462336034],[5.835467498574326,51.39680135084965],[5.833336100034158,51.397377488006626],[5.832700218512237,51.397707787534308],[5.832291014021959,51.398043069787889],[5.832020772294518,51.398428355381782],[5.831896920795533,51.398839616813248],[5.831876615727783,51.399589657946187],[5.821695780960089,51.402101499539263],[5.821697229240763,51.402233881282754],[5.820913411261378,51.402433439876638],[5.820844968254453,51.402334341281446],[5.820704838566742,51.402343572734502],[5.815670770199326,51.403758967491349],[5.815319759910431,51.403871898297936],[5.814139685120287,51.404781264530925],[5.811741949017226,51.405505064038778],[5.809587921429611,51.405703335810465],[5.804269360774491,51.407585803314426],[5.799386078178157,51.408996860056433],[5.787882119254961,51.412647319907947],[5.787608382251762,51.412829519386946],[5.787991123873638,51.412852257988348],[5.787835641907107,51.413037547968457],[5.788120467713253,51.413073377734527],[5.7878275155653,51.413223436993725],[5.787702559815302,51.413406734221894],[5.787355108417315,51.413531990703653],[5.787050473166486,51.413397934665966],[5.787015696092142,51.413675915241257],[5.786721770627751,51.413663305021977],[5.786643392527344,51.413394988778371],[5.786435025686993,51.413654831275849],[5.786203610679678,51.413607219044387],[5.786042823858112,51.413724560571033],[5.786284550778435,51.41388273580769],[5.786489024014438,51.413829648944066],[5.786571781189137,51.413961221897537],[5.786430267386318,51.414091342892711],[5.786097202944718,51.414074451643053],[5.785582821219862,51.414368858565929],[5.785695721878376,51.41459922369615],[5.786032964865821,51.414822169647849],[5.785821106995026,51.414946966490938],[5.785604177215326,51.415259654819025],[5.78525758332557,51.415001449603466],[5.785110961025671,51.415317816003878],[5.784959659110755,51.415157487119608],[5.784843428771405,51.415187798337392],[5.784942791927299,51.415641177093107],[5.784630460934244,51.415551851191864],[5.784416463046251,51.41539398254357],[5.784297893695324,51.415451276518716],[5.784371884809992,51.415634692211384],[5.784045608976719,51.415890176964552],[5.784096029486771,51.416057673037443],[5.783893295950773,51.416117212681392],[5.783777224126541,51.415979414106161],[5.783501975845643,51.415991981682808],[5.783582254298845,51.416253965400401],[5.78333510283959,51.416398248514206],[5.783743977810774,51.416442458412362],[5.78360178140279,51.416604991783984],[5.782913115593389,51.416621947351054],[5.782826899689791,51.416697386194201],[5.78299801637785,51.416827151617618],[5.782960449775488,51.416931054630815],[5.7825818838052,51.416853085188151],[5.782414868274076,51.417051917068306],[5.78211056274063,51.416905514408825],[5.781900180569996,51.41694078879874],[5.782130817024627,51.417197401354954],[5.781903459427148,51.417452194306783],[5.781704367014733,51.417299007317091],[5.781591021382065,51.417324658310072],[5.781759775986676,51.417595188829004],[5.781327633254176,51.417629172106579],[5.781194906769688,51.4178093244469],[5.781025886937272,51.417857167573715],[5.780687270494379,51.417603110648344],[5.780608086172343,51.41778432202296],[5.780751876644331,51.417879648058928],[5.780367368242532,51.418236478693686],[5.780165493039179,51.418222184085593],[5.779941734591396,51.418059045769894],[5.779848980242029,51.418065587454166],[5.779853390081644,51.418176970527121],[5.77975896764251,51.418170133467342],[5.779597348690687,51.41794416040478],[5.779523928308382,51.418082430211193],[5.779722203885425,51.418379920509523],[5.779661066176503,51.418437070263536],[5.779195026269149,51.418442242832676],[5.779216427841082,51.418229487395813],[5.778971702200931,51.418127002096497],[5.778924311192645,51.418351862819343],[5.778564937448999,51.418153959479412],[5.778406692356542,51.418380630305506],[5.778300212062665,51.418384286476631],[5.778210815614148,51.418214284993986],[5.777918631623636,51.418381089643383],[5.777995918588788,51.418644498369332],[5.777743159752112,51.418549012716525],[5.777721531738833,51.418727773067616],[5.777615226872633,51.418782214612001],[5.777206320114551,51.418783071006509],[5.776993554483412,51.418640294802223],[5.776624883741316,51.418891378998822],[5.776730932442282,51.419050683326418],[5.776654574959261,51.419095459858674],[5.776289516114729,51.41884917294756],[5.776005493160262,51.418834740799078],[5.776092605946046,51.419103529226298],[5.775953512185194,51.419246321439708],[5.775583313545879,51.419369973536043],[5.775453727338157,51.41961148415492],[5.774878012911845,51.419758712703448],[5.774629542602294,51.420137193129101],[5.774314437539334,51.420199795062047],[5.77373392368461,51.420594377964974],[5.773164838940988,51.420679454609484],[5.773372835652105,51.420901058152396],[5.773358109083663,51.421138024040324],[5.771469354227158,51.423240662597635],[5.770942035103631,51.423632206596899],[5.770098331165606,51.424036536064492],[5.767085197973971,51.424897081536145],[5.766245139554966,51.425276418270101],[5.765683403452717,51.425768472697271],[5.762789138309887,51.430176359235823],[5.762414881991688,51.430574921556591],[5.761440410610827,51.431137541090294],[5.759168704208082,51.431787677946097],[5.758416625315584,51.43209542308982],[5.757789201458568,51.432546542438409],[5.756949034185631,51.433612562135217],[5.75650026057555,51.433915008971425],[5.755698951246309,51.434267872101813],[5.752627833397639,51.435199766222397],[5.743954309204613,51.437367253549645],[5.73888565714172,51.437900843913987],[5.734550629467962,51.43900976442022],[5.733235931258889,51.439074473267048],[5.721777712830281,51.438218900841314],[5.720871488842447,51.438265839156344],[5.718245546405569,51.438733646636123],[5.718431547735437,51.441218648889226],[5.71824217040778,51.442056397858153],[5.717759164669011,51.44283565353836],[5.718179252938351,51.444533531150611],[5.729301225720664,51.444282026895998],[5.72922462460449,51.444373199685558],[5.730001596630479,51.444355691294874],[5.726226397419075,51.448858149536434],[5.722825317558063,51.453172818657812],[5.724072351265283,51.454127952747498],[5.722703209015932,51.454550172127348],[5.721949883683292,51.454451923628469],[5.720159547938743,51.457452992846939],[5.718827174256673,51.460377227046109],[5.717931543790716,51.462998608865576],[5.717322700934778,51.465658875073629],[5.717186517391878,51.467164853768899],[5.717348544292655,51.467142310011155],[5.717215124288471,51.468238026446393],[5.722973502654876,51.467290909944587],[5.723957445845206,51.470086048162514],[5.735015196813218,51.469186955088787],[5.736093687025283,51.471760111169282],[5.736041319454762,51.472087799063033],[5.735801294674705,51.472381629976681],[5.733738080859943,51.473643331241469],[5.730453287817213,51.47690351762693],[5.728081256546293,51.47884360474314],[5.72637849655227,51.479928795686774],[5.724839971988223,51.481602139584588],[5.722942324338138,51.482888471788371],[5.721717785092947,51.485131464494842],[5.722344209960514,51.484669768297167],[5.723356050446312,51.484244927300146],[5.724056947914847,51.484096066263859],[5.724827433159065,51.484113857218894],[5.725730168604042,51.48441037710279],[5.728617274730899,51.486094351774639],[5.729411936733566,51.486249470295022],[5.730428626750109,51.486217576449697],[5.73179324608951,51.486032270025682],[5.732474457792871,51.485805008010544],[5.734461437310184,51.484511486482958],[5.735045575531773,51.484330862047443],[5.735608715086956,51.484303887653319],[5.736239944414102,51.484429937609725],[5.737397944913579,51.484943202530857],[5.738729057240374,51.485272456829939],[5.745948253238707,51.485886688905921],[5.748063437168574,51.486143428548083],[5.749349365632382,51.486378356023138],[5.750721052374785,51.486907877372062],[5.75189121411058,51.487243155048496],[5.752323217356885,51.487287961231473],[5.754220533482063,51.48723946553288],[5.754588621366609,51.487319160833749],[5.755562140286128,51.48775840007557],[5.757635409596562,51.488068123368549],[5.758297408090398,51.488079700866521],[5.759856267199219,51.487834113847285],[5.762089068453235,51.488131329495424],[5.763230200742568,51.487922831746197],[5.764043279703675,51.487898321319783],[5.764740569191813,51.488035060206002],[5.766670148791788,51.488776097668691],[5.768003837060713,51.488906341461245],[5.768418631877835,51.489020896861867],[5.769523772732469,51.488880445935933],[5.773439452214885,51.488638437768813],[5.773961211158424,51.488665284883446],[5.778472244584662,51.490202916389997],[5.780621481941387,51.490141911012593],[5.784285738321734,51.49048494216472],[5.786502075027313,51.490567689702743],[5.789207352554276,51.490957376287525],[5.791385308110553,51.491437375868138],[5.792935773635736,51.491697469257005],[5.793464808336839,51.491859745251567],[5.794417638694719,51.49269020213849],[5.794727084243867,51.492831710401525],[5.796954615815005,51.493584250573718],[5.799028130001587,51.49409701387912],[5.799796790579556,51.494185103280152],[5.800986711862857,51.493871300227745],[5.801338069983016,51.493895213467113],[5.814709847958937,51.496627173731618],[5.852573782872571,51.504144422431153],[5.860723723446023,51.491966547944592],[5.871695276560999,51.450111962664977],[5.874965974060582,51.446557719307457],[5.874965974060582,51.446557719307457]]]}
}
,{
"type":"Feature","properties":{
"gemeentena":"Onderbanken"}
,"geometry":{
"type":"Polygon","coordinates":[[[6.006053605132877,50.957535393339093],[6.002401173208973,50.957880466196471],[6.000792857863228,50.957889359991029],[5.999893976469209,50.957796858965274],[5.999249666254875,50.957829175260315],[5.999242261992614,50.957775226431188],[5.993381929070956,50.958762647414552],[5.98012353883364,50.960679643943841],[5.978038638591675,50.959881552768103],[5.977895442268708,50.959988731007186],[5.977093899865662,50.959579072045429],[5.976855959754464,50.959333482293559],[5.974728528775707,50.958393873084006],[5.974022559429245,50.958340552920063],[5.973957260044799,50.959374617513305],[5.972300627118779,50.959274602526953],[5.969083882248002,50.961035574889543],[5.970153876524136,50.961929531925868],[5.970680216742294,50.962142436758313],[5.96999025596251,50.962502324739887],[5.965861675852519,50.962417679649576],[5.965521114638721,50.961160770644675],[5.965356964626454,50.960924811590907],[5.962755248464621,50.958739553837269],[5.959185243316178,50.956959521056667],[5.958387847640435,50.956674066431063],[5.957428820117979,50.955817415915433],[5.95568678432673,50.954915614969792],[5.953232376347503,50.953363370101357],[5.952499330643874,50.952784645954367],[5.952673340544478,50.952470179876599],[5.949294007460875,50.951557140532159],[5.949598170670297,50.951207412639562],[5.949572566159977,50.951114761843634],[5.94555608747551,50.949891630454488],[5.943800087430192,50.949154640216683],[5.942642717945735,50.949505492107917],[5.942292261577371,50.949014083254255],[5.938414119597067,50.949918862430707],[5.930709241304757,50.951472124399181],[5.927495434470557,50.95240708848636],[5.925545617039077,50.953164214555358],[5.917601664554483,50.956835554813949],[5.918408164417021,50.957325620497663],[5.921751102998452,50.958577578418627],[5.920450829974452,50.959262285216518],[5.918963833025144,50.959757637752439],[5.918713496359853,50.959998450660585],[5.922724922434719,50.962913764068503],[5.921174135076361,50.963761276391367],[5.917521646840373,50.967629423943663],[5.915793859301596,50.967414235984059],[5.914339555470781,50.967609660380724],[5.912797733245119,50.967988928220493],[5.915255289053706,50.969961051893037],[5.916005434012722,50.969965296403544],[5.916256468884949,50.970372077439855],[5.916277375902652,50.970566565773879],[5.915446172239114,50.970613282598535],[5.91405227897282,50.971203563802668],[5.915025904732275,50.971871665588104],[5.914541393956014,50.972156988919743],[5.914459961745558,50.972347997386223],[5.915828047585506,50.972363486791522],[5.918363937868452,50.972935401263648],[5.91982071105365,50.973543105518679],[5.924221995762851,50.975694943400363],[5.922464293281297,50.976629527310216],[5.922413853143246,50.976555170059285],[5.920346728850657,50.977690310425182],[5.920247763445007,50.977612067656175],[5.919655802179189,50.97819316700096],[5.922993111356025,50.979545724470064],[5.927651528028723,50.982285399099034],[5.93388644645889,50.985353196074243],[5.950134484551962,50.98720719083223],[5.955185081844721,50.988416992002058],[5.95739231127415,50.985005257689316],[5.960879579524033,50.983155724619351],[5.960786346543608,50.983055959210226],[5.965010889184503,50.980939539799536],[5.965844851125889,50.980082950995204],[5.966325913791048,50.979769437855794],[5.966911339673838,50.979534952903641],[5.967575848085406,50.979395891308386],[5.968508911310305,50.979426988176577],[5.973937145974041,50.980553524301584],[5.979575320451537,50.982916266288484],[5.981357232493251,50.983217259743043],[5.981817446499117,50.983192245833621],[5.982243219122747,50.983068348299703],[5.984222245955424,50.982114580541776],[6.017363527634932,50.983401717666851],[6.017226181033721,50.983678418749228],[6.018700580149773,50.983818537888219],[6.021199563158425,50.983593554761839],[6.022930748530938,50.983269711001988],[6.025691941066555,50.983203744367252],[6.026511843791895,50.983272024042947],[6.02679073965514,50.981674797157893],[6.026894541966229,50.981641378017059],[6.026802350902174,50.981309383009219],[6.022645336168819,50.974020492535729],[6.021163202467538,50.971002919114056],[6.01945513614732,50.968948638232092],[6.01878941486312,50.967947568072383],[6.017307849263054,50.965273585276542],[6.015207061556862,50.962364384683774],[6.013000479066401,50.960397758743611],[6.010687440444448,50.960031700286763],[6.009080817370868,50.959484729347103],[6.00773695727333,50.958820010535447],[6.006640438382172,50.958056152864494],[6.006053605132877,50.957535393339093],[6.006053605132877,50.957535393339093]]]}
}
,{
"type":"Feature","properties":{
"gemeentena":"Sliedrecht","Aantal":5.1}
,"geometry":{
"type":"Polygon","coordinates":[[[4.803378619488583,51.84635340433428],[4.803904108138668,51.838648216625948],[4.804116681566933,51.838296139576514],[4.804564910766739,51.837910077638973],[4.804692638455397,51.837897938707904],[4.804684134879049,51.837646612749438],[4.804128889434624,51.837307981538025],[4.804005493608753,51.836820791664849],[4.804673813757646,51.827586800245022],[4.804735055887873,51.824937034666398],[4.80494641231187,51.823901082141809],[4.804961658725929,51.823203654472948],[4.805078937065509,51.823196577901847],[4.805075519908264,51.821949425033246],[4.808170365112693,51.82180480856038],[4.811029973852395,51.821060596690771],[4.811384174012619,51.821147878964396],[4.811415902936563,51.818658194537399],[4.807978322464671,51.81846478104309],[4.805248542936508,51.818001768630076],[4.800167504980088,51.81756833926994],[4.796208891641041,51.816952386704038],[4.788816276972459,51.815573868716648],[4.786803679118083,51.815313671606354],[4.78535495685803,51.815198388408923],[4.782539540011199,51.815132610109096],[4.777198780382886,51.815324689593638],[4.772091669022918,51.815319085072375],[4.767944209226318,51.815719019335681],[4.765035469187072,51.816350086398458],[4.762117637465085,51.817307714778075],[4.757745547141138,51.819196258853019],[4.754617838331463,51.820683640362155],[4.752268550681958,51.821409512495769],[4.749798433214361,51.821926191089247],[4.747395604487978,51.822221166179745],[4.74563122374044,51.822291965263851],[4.743032362296625,51.822296211534592],[4.735948172606505,51.822033454554713],[4.729638741001389,51.822168736516993],[4.73215200327967,51.82574672993033],[4.741410361766707,51.840776403562053],[4.74150891746398,51.840834256855615],[4.741970855969564,51.840800887020137],[4.742764006541115,51.841002686526636],[4.745305120499263,51.841733698189401],[4.746211574905002,51.842073313357098],[4.74724606947518,51.842681074659566],[4.748792733463373,51.843201061701961],[4.751725686606761,51.843926827415025],[4.753499042060125,51.844447367480115],[4.756128814856329,51.844957736429976],[4.758081091107573,51.844623546347393],[4.760018581623526,51.843900426336219],[4.761249583769639,51.843743367543134],[4.762596135341354,51.843778798578306],[4.765225767963789,51.844256523050504],[4.766524029404535,51.843898951714422],[4.768013981243888,51.84386255878627],[4.770455948067382,51.844323329218064],[4.771870362973537,51.844696863254953],[4.773142437866078,51.844710470968167],[4.774699661038295,51.845591798041553],[4.778623436957229,51.846393620551311],[4.780379803916428,51.846639927448813],[4.782210545716854,51.846340069333387],[4.78506681920978,51.845612404856396],[4.7858405269155,51.845489928036947],[4.786513979167425,51.845120440291417],[4.786901042592209,51.845005478771604],[4.788147396815892,51.844937418998782],[4.795601318557641,51.844975697015137],[4.797913830666958,51.845363900583244],[4.802177487839542,51.846241451244026],[4.803378619488583,51.84635340433428],[4.803378619488583,51.84635340433428]]]}
}
```

**Comment by tomchadwin** [comment]
I don't understand this yet. It looks like it's a problem with how the layer data export routine handles `null`s, but if so, I'm surprised not to have seen the issue before.

**Comment by tomchadwin** [comment]
Are you able to upload your QGIS layer?

**Comment by tomchadwin** [comment]
And ideally export and upload your QGIS style for that layer.

**Comment by Jeroen-GroeneBij** [comment]
I will try.
Are the following files necessary?
.shp 
.dbf
.shx
.qgs
.qgs~

I can put them in a zip.
Or is there a better way to export the project?

**Comment by Jeroen-GroeneBij** [comment]
and probably the .qml for the style?

**Comment by tomchadwin** [comment]
Not the last one ending in ~, but the others, yes. I was in the middle of telling you how to export your .qml a well, but plainly you know how to do that. Yes, we need that as well.

**Comment by Jeroen-GroeneBij** [comment]
[Laadpuntendichtheid.zip](https://github.com/tomchadwin/qgis2web/files/1252346/Laadpuntendichtheid.zip)

**Comment by tomchadwin** [comment]
I'll have to continue looking into this tomorrow. Sorry not to have a solution yet.

**Comment by Jeroen-GroeneBij** [comment]
No problem, I'll check in occasionally this weekend and later on.
Thank you all ready.

**Comment by tomchadwin** [comment]
OK, I've got to the bottom of this, and it;s a bit more complex than I was anticipating. I finally remembered that I'd heard of a similar issue, and tracked it down to here:

https://github.com/UCDavisLibrary/ava/issues/270

This was logged as a QGIS bug:

https://issues.qgis.org/issues/16812

In fact, it's an underlying change in how OGR works:

https://trac.osgeo.org/gdal/wiki/rfc67_nullfieldvalues

It was fixed in QGIS, and released in QGIS 2.18.11 and later versions. However, you are (very reasonably) using QGIS 2.14, the long-term release (LTR). I am assuming that the fix was not backported to 2.14. I've asked specifically here:

http://osgeo-org.1560.x6.nabble.com/GeoJSON-null-fix-in-LTR-td5332616.html

In the meantime, I guess you have a couple of options:

- upgrade to QGIS 2.18.12
- change that value to something non-null and change your styling rule accordingly

Sorry not to have an easier fix. I'll close this issue, but feel free to reopen if anything else remains relating to this problem.

**Comment by Jeroen-GroeneBij** [comment]
Hello Tom,

 

Thank you very much for looking into this.

For now, I’ll try to replace null values to something else. I just have to find a suitable value, because they will show up in ‘popup on hover’ , so a 0 or 999 value in a numeric attribute gives the wrong message. I do have the change the field type because I can’t enter text in a numeric field type.

 

Or is there a way, like alias, that will change a numeric 999 into a word  or “-“ when viewed in the popup?

 

Greetings

Jeroen Hovens

 

 

 

Van: Tom Chadwin [mailto:notifications@github.com] 
Verzonden: zaterdag 26 augustus 2017 12:47
Aan: tomchadwin/qgis2web <qgis2web@noreply.github.com>
CC: Jeroen-GroeneBij <mail@groenebij.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [tomchadwin/qgis2web] No map display with style rule for NULL value (#533)

 

OK, I've got to the bottom of this, and it;s a bit more complex than I was anticipating. I finally remembered that I'd heard of a similar issue, and tracked it down to here:

UCDavisLibrary/ava#270 <https://github.com/UCDavisLibrary/ava/issues/270> 

This was logged as a QGIS bug:

https://issues.qgis.org/issues/16812

In fact, it's an underlying change in how OGR works:

https://trac.osgeo.org/gdal/wiki/rfc67_nullfieldvalues

It was fixed in QGIS, and released in QGIS 2.18.11 and later versions. However, you are (very reasonably) using QGIS 2.14, the long-term release (LTR). I am assuming that the fix was not backported to 2.14. I've asked specifically here:

http://osgeo-org.1560.x6.nabble.com/GeoJSON-null-fix-in-LTR-td5332616.html

In the meantime, I guess you have a couple of options:

*	upgrade to QGIS 2.18.12
*	change that value to something non-null and change your styling rule accordingly

Sorry not to have an easier fix. I'll close this issue, but feel free to reopen if anything else remains relating to this problem.

**Comment by tomchadwin** [comment]
Are you sure changing the field type to text is not an option? If you could, then you could just set that record's value to a space character.

**Comment by Jeroen-GroeneBij** [comment]
It is indeed as simple as that. Creating a new attribute, string type, filling it with the obsolete attribute values, changing the 'null' to something else, and deleting the obsolete attribute and adjusting the style rules.
To bad I can't just edit an existing field type. Maybe that is an option in 2.18?

Thanks again and until next time.

**Comment by tomchadwin** [comment]
I looked this up, and if I am reading the roadmap properly, this should be fixed in 2.14.17. Are you able to upgrade to the latest LTR point release and retest? Let me know.

**Comment by Jeroen-GroeneBij** [comment]
Good news,
in qgis 2.14.18 and qgis2web 2.24.0 the NULL value problem is gone.
A rule based style with "Attribute" IS NULL works as expected and the webmap shows all the fields with a NULL value.

**Comment by tomchadwin** [comment]
Brilliant - many thanks for investigating and confirming.

**Comment by Jeroen-GroeneBij** [comment]
One addition:
Using  Style > Category and then classify brings back the problem.
Changing Category to Rule-Based shows why: Category classifies the NULL values as ELSE and somehow this doesn't work well with qgis2web.

So I am forced to use Rule-Based and use "Attribute" IS NULL
Which is no problem for me.

Maybe this will help you understand the issue better and come up with a even more suitable solution.

**Comment by Jeroen-GroeneBij** [comment]
Sorry to re-open this again.

My good news isn't all true.

I have found another issue, this time between different browsers:
The map made with qgis 2.1.4.18 has a problem when viewed in Chrome, however, when viewed in Firefox (or Edge or IE) there is no problem.
See [https://www.groenebij.nl/Kaarten/GDAW](url)

This is the js console error form Chrome:
Uncaught ReferenceError: DeelnamegemeentenGDAW0rule0_eval_expression is not defined
    at style_DeelnamegemeentenGDAW0_0 (www.groenebij.nl/:76)
    at e._setLayerStyle (leaflet.js:8)
    at e.resetStyle (leaflet.js:8)
    at e.addData (leaflet.js:8)
    at e.addData (leaflet.js:8)
    at e.initialize (leaflet.js:8)
    at new e (leaflet.js:5)
    at new o.geoJSON (leaflet.js:8)
    at www.groenebij.nl/:122

The other map, made with qgis 2.14.16 (the one I started this issue with), has no problem in both Chrome and Firefox. There the NULL issue was resolved by changing the attribute type and using an appropriate value. 
See [https://www.groenebij.nl/Kaarten/Laadpunten](url)

Let me know if I need to change something, do some further testing etc.
